src/Microsoft.AspNet.Mvc.Abstractions/ModelBinding/ModelStateDictionary.cs

```diff
@@ -657,8 +657,19 @@ public bool TryGetValue(string key, out ModelStateEntry value)
             return _innerDictionary.TryGetValue(key, out value);
         }
 
+        /// <summary>
+        /// Returns an enumerator that iterates through the <see cref="ModelStateDictionary"/>.
+        /// </summary>
+        /// <returns>
+        /// A <see cref="ModelStateDictionary.Enumerator"/> structure for the <see cref="ModelStateDictionary"/>.
+        /// </returns>
+        public Enumerator GetEnumerator()
+        {
+            return new Enumerator(_innerDictionary.GetEnumerator());
```

Comment:
That enumerator is a struct, we just don't want to expose Dictionary<T,K>.Enumerator directly as it's leaking an implementation detail